File: ring-android/libringclient/src/main/java/cx/ring/model/AccountCredentials.java
/**
 * Copyright (C) 2004-2016 Savoir-faire Linux Inc.
 * <p>
 * Author: Alexandre Lision <alexandre.lision@savoirfairelinux.com>
 * Adrien Béraud <adrien.beraud@savoirfairelinux.com>
 * <p>
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * <p>
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * <p>
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package cx.ring.model;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

public class AccountCredentials implements Serializable {

    @SuppressWarnings("unused")
    private static final String TAG = AccountCredentials.class.getSimpleName();

    private String mUsername;
    private String mPassword;
    private String mRealm;

    public AccountCredentials(Map<String, String> pref) {
        mUsername = pref.get(ConfigKey.ACCOUNT_USERNAME.key());
        mPassword = pref.get(ConfigKey.ACCOUNT_PASSWORD.key());
        mRealm = pref.get(ConfigKey.ACCOUNT_REALM.key());
    }

    public AccountCredentials(String username, String password, String realm) {
        setUsername(username);
        setPassword(password);
        setRealm(realm);
    }

    public void setUsername(String val) {
        mUsername = val;
    }

    public void setPassword(String val) {
        mPassword = val;
    }

    public void setRealm(String val) {
        mRealm = val;
    }

    public String getUsername() {
        return mUsername;
    }

    public String getPassword() {
        return mPassword;
    }

    public String getRealm() {
        return mRealm;
    }

    public HashMap<String, String> getDetails() {
        HashMap<String, String> details = new HashMap<>();
        details.put(ConfigKey.ACCOUNT_USERNAME.key(), mUsername);
        details.put(ConfigKey.ACCOUNT_PASSWORD.key(), mPassword);
        details.put(ConfigKey.ACCOUNT_REALM.key(), mRealm);
        return details;
    }

    public void setDetail(ConfigKey key, String value) {

        switch (key) {
            case ACCOUNT_USERNAME:
                mUsername = value;
                break;
            case ACCOUNT_PASSWORD:
                mPassword = value;
                break;
            case ACCOUNT_REALM:
                mRealm = value;
                break;
        }
    }
}

File: jami-android/libjamiclient/src/main/kotlin/net/jami/model/AccountCredentials.kt
/*
 *  Copyright (C) 2004-2025 Savoir-faire Linux Inc.
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program. If not, see <https://www.gnu.org/licenses/>.
 */
package net.jami.model

import java.io.Serializable
import java.util.HashMap

class AccountCredentials : Serializable {
    var username: String? = null
    var password: String? = null
    var realm: String? = null

    constructor(pref: Map<String, String>) {
        username = pref[ConfigKey.ACCOUNT_USERNAME.key]
        password = pref[ConfigKey.ACCOUNT_PASSWORD.key]
        realm = pref[ConfigKey.ACCOUNT_REALM.key]
    }

    constructor(username: String?, password: String?, realm: String?) {
        this.username = username
        this.password = password
        this.realm = realm
    }

    val details: HashMap<String, String>
        get() {
            val details = HashMap<String, String>()
            details[ConfigKey.ACCOUNT_USERNAME.key] = username ?: ""
            details[ConfigKey.ACCOUNT_PASSWORD.key] = password ?: ""
            details[ConfigKey.ACCOUNT_REALM.key] = realm ?: ""
            return details
        }

    fun setDetail(key: ConfigKey, value: String?) {
        when (key) {
            ConfigKey.ACCOUNT_USERNAME -> username = value
            ConfigKey.ACCOUNT_PASSWORD -> password = value
            ConfigKey.ACCOUNT_REALM -> realm = value
            else -> {}
        }
    }

    companion object {
        private val TAG = AccountCredentials::class.java.simpleName
    }
}